Q1. Write a program that prints number from 1 to 10 number with 
their square.

In [10]:
def sequare(n):
    i=1
    while i<=n:
        print(i**2,end=',')
        i+=1
sequare(int(input()))

1,4,9,16,25,36,49,64,81,100,

Q2. Write a program that prints a table of a number, take that 
number from the user.
Sample input:
Enter a number: 5
Output:
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50

In [16]:
def table(n):
    i=1
    while i<=10:
        print(f'{n} * {i} = {i*n}')
        i+=1
table(int(input()))

5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


Q3. Write a program that calculates the product of all numbers 
between 1 to 10 using a while loop.

In [26]:
def product_sequance(n):
    i=1
    p=1
    while i<=n:
        p*=i
        i+=1
    return p
product_sequance(int(input()))


120

Q4. Write a program that prints numbers divisible of 7 between 1 to 
70 

In [29]:
def divisibal_7(n):
    i=1
    while i<=n:
        if i%7==0:
            print(i,end=',')
        i+=1
divisibal_7(int(input()))

7,14,21,28,35,42,49,56,63,70,

Q5. Write a program that takes input from the user and print the 
reverse counting of that number.
Sample Input: 5
Sample output: 5 4 3 2 1
Note: Write the dry run and generate output of all the below 
programs by using paper and pen in your notebook.

In [7]:
def reverse(n):
    while n>=1:
        print(n,end=',')
        n-=1
reverse(int(input()))

5,4,3,2,1,

assignment 2

Q10. Write a program to print Multiples of 3 and 5 between 1 to 30.
Output: 15, 30


In [14]:
def multiples(n):
    i=1
    while i<=n:
        if i%3==0 and i%5==0:
            print(i,end=',')
        i+=1
multiples(int(input()))

15,30,

Q11. Write a program to calculate the sum of numbers from 1 to 10 using a while
loop.
Output: 55

In [15]:
def sum(n):
    i=1
    s=0
    while i<=n:
        s+=i
        i+=1
    return s
sum(int(input()))


55

Q12. Write a program to calculate Sum of Odd Numbers up to 30 using while loop

In [16]:
def sum_odd(n):
    i=1
    while i<=n:
        if i%2!=0:
            print(i,end=',')
        i+=1
sum_odd(int(input()))

1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,

assignment-3

Q1. Write a python program to find the Sum of the Digits of a given 
number as an input.
Input Output Explanation
123 6 1+2+3 = 6
456 15 4+5+6 = 15
789 24 7+8+9 = 24

In [17]:
def sum_digit(n):
    i=1
    s=0
    while i<=3:
        s+=n%10
        n//=10
        i+=1
    return s
sum_digit(int(input()))


6

Q2. Write a python program which finds the reverse of a given 
number.
Input Output Explanation
123 321 Reverse of 123 is 321.
456 654 Reverse of 456 is 654.
789 987 Reverse of 789 is 987.


In [18]:
def revers_digit(n):
    i=1
    s=0
    while i<=3:
        s=s*10+n%10
        n//=10
        i+=1
    return s
revers_digit(int(input()))


321

Q3. Write a python program which find the Factors of a given number
as an input.
Explanation: Given an integer input, the objective is to find all the factors
ofthe number by checking divisibility.
Input Output Explanation
12 1, 2, 3, 4, 6, 12 Factors of 12 are 1, 2, 3, 4, 6, and 12.
15 1, 3, 5, 15 Factors of 15 are 1, 3, 5, and 15.
18 1, 2, 3, 6, 9, 18 Factors of 18 are 1, 2, 3, 6, 9, and 18.


In [19]:
def fact(n):
    i=1
    while i<=n:
        if n%i==0:
            print(i,end=',')
        i+=1
fact(int(input()))
    


1,2,3,4,6,12,

Q5. Write a python program which check Whether the given number
as an input is a Palindrome or not.
Explanation: palindrome numbers are those whose reverse is equal to the 
original number.
Input Output Explanation
121 Palindrome Reverse of 121 is 121, so it’s a palindrome.
123 Not a Palindrome Reverse of 123 is 321, so it’s not a palindrome.
1331 Palindrome Reverse of 1331 is 1331, so it’s a palindrome

In [22]:
def is_palindrome(n):
    s=0
    x=n
    i=1
    while i<=3:
        s=s*10+n%10
        n//=10
        i+=1
    if s==x:
        print(True)
    else:
        print(False)
is_palindrome(int(input()))

True


Q6. Write a python program which finds Factorial of a Number.
Input Output Explanation
5 120 Factorial of 5 is 5! = 5×4×3×2×1 = 120.
6 720 Factorial of 6 is 6! = 6×5×4×3×2×1 = 720.
0 1 Factorial of 0 is 1 (by definition).


In [23]:
def factoral(n):
    p=1
    i=1
    while i<=n:
        p*=i
        i+=1
    return p
factoral(int(input()))

120

Q7. Check Whether or Not the Number is an Armstrong Number.
Explanation: An Armstrong number is a number that equals the sum of its 
digits, each raised to a power (length of that number or count of digit in that 
number).
Input Output Explanation
153 Armstrong 1³ + 5³ + 3³ = 153.
123 Not Armstrong 1³ + 2³ + 3³ ≠ 123.
370 Armstrong 3³ + 7³ + 0³ = 370.

In [25]:
def is_armstrong(n):
    x=n
    s=0
    i=1
    while i<=3:
        r=n%10
        s+=r**3
        n//=10
        i+=1
    if s==x:
        print(True)
    else:
        print(False)
is_armstrong(int(input()))



False
